apps/web-giddh/src/app/vouchers/preview/preview.component.ts

```diff
@@ -1,90 +1,1170 @@
+import { Store } from "@ngrx/store";
+import { InvoiceReceiptActions } from "../../actions/invoice/receipt/receipt.actions";
+import { saveAs } from 'file-saver';
+import { Location } from "@angular/common";
+import { NewConfirmationModalComponent } from "../../theme/new-confirmation-modal/confirmation-modal.component";
+import { AdjustAdvancePaymentModal, VoucherAdjustments } from "../../models/api-models/AdvanceReceiptsAdjust";
+import { AdjustmentUtilityService } from "../../shared/advance-receipt-adjustment/services/adjustment-utility.service";
 
 @Component({
     selector: "preview",
     templateUrl: "./preview.component.html",
-    styleUrls: ["./preview.component.scss"]
+    styleUrls: ["./preview.component.scss"],
+    providers: [VoucherComponentStore]
 })
 export class VouchersPreviewComponent implements OnInit, OnDestroy {
+    /** Instance of PDF container iframe */
+    @ViewChild('pdfContainer', { static: false }) pdfContainer: ElementRef;
     /** Instance of cdk scrollbar */
     @ViewChild(CdkVirtualScrollViewport) cdkScrollbar: CdkVirtualScrollViewport;
-    // export dialog
-    @ViewChild('paidDialog', { static: true }) public paidDialog: TemplateRef<any>;
-    // adjust payment dialog
+    /** Instance of Adjust Payment Dialog */
     @ViewChild('adjustPaymentDialog', { static: true }) public adjustPaymentDialog: TemplateRef<any>;
-    // history dialog
+    /** Instance of Version History Dialog */
     @ViewChild('historyAsideDialog', { static: true }) public historyAsideDialog: TemplateRef<any>;
-    // email dialog
-    @ViewChild('emailSendDialog', { static: true }) public emailSendDialog: TemplateRef<any>;
-    /* This will hold local JSON data */
-    @Input() public localeData: any = {};
-    /* This will hold common JSON data */
-    @Input() public commonLocaleData: any = {};
+    /** Holds send email dailog template reference send email */
+    @ViewChild('sendEmailModal', { static: true }) public sendEmailModal: any;
+    /** Holds Payment template reference */
+    @ViewChild('paymentDialog', { static: true }) public paymentDialog: TemplateRef<any>;
+    /** Attached document preview container instance */
+    @ViewChild('attachedDocumentPreview', { static: true }) attachedDocumentPreview: ElementRef;
     /** Observable to unsubscribe all the store listeners to avoid memory leaks */
     private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** Voucher PDF for preview downloading is in progress Observable */
+    public isVoucherDownloading$: Observable<boolean> = this.componentStore.select(state => state.isVoucherDownloading);
+    /** Voucher Download Error in PDF for preview downloading status Observable */
+    public isVoucherDownloadError$: Observable<boolean> = this.componentStore.select(state => state.isVoucherDownloadError);
+    /** Get Vouchers is in progress Observable */
+    public getVouchersInProgress$: Observable<any> = this.componentStore.getLastVouchersInProgress$;
```

Comment:
```suggestion
    public getVouchersInProgress$: Observable<any> = tthis.componentStore.select(state => state. getLastVouchersInProgress);
```

Reply:
Done make all code common like this 
public getVouchersInProgress$: Observable<any> = this.componentStore.getLastVouchersInProgress$